src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -42,97 +42,80 @@ private enum GlobalNotificationState
         private readonly object _globalNotificationsGate = new object();
         private Task<GlobalNotificationState> _globalNotificationsTask = Task.FromResult(GlobalNotificationState.NotStarted);
 
-        public static async Task<RemoteHostClient?> CreateAsync(
-            Workspace workspace, CancellationToken cancellationToken)
+        public static async Task<RemoteHostClient?> CreateAsync(Workspace workspace, CancellationToken cancellationToken)
         {
-            try
+            using (Logger.LogBlock(FunctionId.ServiceHubRemoteHostClient_CreateAsync, cancellationToken))
             {
-                using (Logger.LogBlock(FunctionId.ServiceHubRemoteHostClient_CreateAsync, cancellationToken))
-                {
-                    var primary = new HubClient("ManagedLanguage.IDE.RemoteHostClient");
-                    var timeout = TimeSpan.FromMilliseconds(workspace.Options.GetOption(RemoteHostOptions.RequestServiceTimeoutInMS));
-
-                    // Retry (with timeout) until we can connect to RemoteHost (service hub process). 
-                    // we are seeing cases where we failed to connect to service hub process when a machine is under heavy load.
-                    // (see https://devdiv.visualstudio.com/DevDiv/_workitems/edit/481103 as one of example)
-                    var instance = await Connections.RetryRemoteCallAsync<IOException, ServiceHubRemoteHostClient>(
```

Comment:
❔ IOException no longer needs to be handled?

Reply:
This "fixed" some crash in past for which dumps are no longer available. But I don't think this is a good fix. We shouldn't be doing any retries at all. That logic should be in ServiceHub and in JsonRPC. Checked with ServiceHub - they might need to do some fixing around this, so I'm leaving the retries around ServiceHub call that retrieves the Stream.
JsonRPC's invocation does not declare IOException as an exception that can be thrown. I'll double check with Andrew if that is indeed the case.

Reply:
See https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1040692 tracking ServiecHub exceptions.

Reply:
Works for me thanks!